src/helpers/job.coffee

```diff
@@ -66,12 +67,16 @@ JobListenerConfig = (checkIntervals, checkRepeats) ->
       # whenever this job status is updated, emit the status for this wrapper
       updateJobStatus = @_updateJobStatusFor(id)
       JobStore.on("job.#{jobId}.*", updateJobStatus)
-      JobStore.off("job.#{jobId}.final", updateJobStatus)
+      JobStore.once("job.#{jobId}.final", @stopListening(updateJobStatus))
 
     stopChecking: (id) ->
       jobId = @_getLatestJob(id)
       JobActions.stopChecking(jobId) if jobId?
 
+    stopListening: (updateJobStatus) ->
+      (jobData) ->
+        JobStore.off("job.#{jobData.id}.*", updateJobStatus)
+
```

Comment:
aw darn...looks like i caused a test failure with that, ill fix that tomorrow, esp if you think 94d996d is the way to go